libs/cart/src/drivers/magento/cart-shipping-information.service.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable } from '@angular/core';
+import { Apollo } from 'apollo-angular';
+
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { DaffCartShippingInformationServiceInterface } from '../interfaces/cart-shipping-information-service.interface';
+import { DaffCart } from '../../models/cart';
+import { DaffMagentoCartShippingRateTransformer } from './transforms/outputs/cart-shipping-rate.service';
+import {
+  getSelectedShippingMethod,
+  setSelectedShippingMethod
+} from './queries';
+import { DaffMagentoShippingMethodInputTransformer } from './transforms/inputs/shipping-method.service';
+import { DaffMagentoCartTransformer } from './transforms/outputs/cart.service';
+import { DaffCartShippingRate } from '../../models/cart-shipping-rate';
+import { MagentoGetSelectedShippingMethodResponse, MagentoSetSelectedShippingMethodResponse } from './models/responses';
+
+/**
+ * A service for making Magento GraphQL queries for carts.
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class DaffMagentoCartShippingInformationService implements DaffCartShippingInformationServiceInterface<DaffCartShippingRate, DaffCart> {
+  constructor(
+    private apollo: Apollo,
+    public cartTransformer: DaffMagentoCartTransformer,
+    public shippingRateTransformer: DaffMagentoCartShippingRateTransformer,
+    public shippingMethodInputTransformer: DaffMagentoShippingMethodInputTransformer
+  ) {}
+
+  get(cartId: string): Observable<DaffCartShippingRate> {
+    return this.apollo.query<MagentoGetSelectedShippingMethodResponse>({
+      query: getSelectedShippingMethod,
+      variables: {cartId}
+    }).pipe(
+      map(result => this.shippingRateTransformer.transform(result.data.cart.selected_shipping_method))
+    );
+  }
+
+  update(cartId: string, shippingInfo: Partial<DaffCartShippingRate>): Observable<Partial<DaffCart>> {
+    return this.apollo.mutate<MagentoSetSelectedShippingMethodResponse>({
+      mutation: setSelectedShippingMethod,
+      variables: {
+        cartId,
+        method: this.shippingMethodInputTransformer.transform(shippingInfo)
+      }
+    }).pipe(
+      map(result => this.cartTransformer.transform(result.data.setShippingMethodsOnCart.cart))
+    )
+  }
+
+  delete(cartId: string, id?: string | number): Observable<Partial<DaffCart>> {
+    return this.apollo.mutate<MagentoSetSelectedShippingMethodResponse>({
+      mutation: setSelectedShippingMethod,
+      variables: {
+        cartId,
+        method: {
+          carrier_code: '',
+          method_code: ''
+        }
+      }
+    }).pipe(
+      map(result => this.cartTransformer.transform(result.data.setShippingMethodsOnCart.cart))
```

Comment:
why do the daffodil CartShippingInformation and SelectPaymentOption driver interfaces differ here? In this delete method, the return is a DaffCart, but for the set payment option driver, the remove method returned void.

Reply:
I believe the rationale was that removing the payment didn't need to return the cart because it wouldn't affect other parts of the cart; it was sufficient just to set `payment` to `null`. However removing shipping info could affect totals, discounts, etc so the entire cart must be returned to be safe.

Reply:
gotcha. makes sense